map/framework.cpp

```diff
@@ -517,10 +517,16 @@ bool Framework::OnCountryFileDelete(storage::TCountryId const & countryId, stora
 
 void Framework::OnMapDeregistered(platform::LocalCountryFile const & localFile)
```

Comment:
Gui - неудачное название. Правильное название - main 

> On Mar 25, 2016, at 23:13, ygorshenin <notifications@github.com> wrote:
> 
> In map/framework.cpp <https://github.com/mapsme/omim/pull/2523#discussion_r57465062>:
> 
> > @@ -517,10 +517,16 @@ bool Framework::OnCountryFileDelete(storage::TCountryId const & countryId, stora
> >  
> >  void Framework::OnMapDeregistered(platform::LocalCountryFile const & localFile)
> Потому что MwmSet может быть создан где угодно, в том числе и в тестах, где нет понятия GuiThread. Посему он и не знает про GuiThread. И поэтому из двух возможностей:
> 1. Звать каллбек на GuiThread
> 2. Звать каллбек с любого потока а каллбек внутри себя уже решит сам куда дальше запостить задачу
> 
> MwmSet выбирает второй подход, так как он гибче.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2523/files/8bff1dbc61d252545ea4b21bffd137f33f43a0e2#r57465062>